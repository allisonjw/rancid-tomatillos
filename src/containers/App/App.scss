$dark-grey-color: #777;
$color-white: #ffffff;
$color-black: #000;
$color-dark-grey-2: #999;
$primary-color: #7ed56f;
$color-secondary-light: #ffb900;
$color-secondary-dark: #ff7730;
$color-card-light: #2998ff;
$color-card-dark: #1000a2;
$font-Bebas: 'Bebas Neue', cursive;

@import url('https://fonts.googleapis.com/css?family=Rubik:300,400,500,700,900&display=swap');
@import url('https://fonts.googleapis.com/css?family=Bebas+Neue&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
html {
  font-size:62.5%
}


.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.btn-grey {
  font-weight: 700;
  color: $color-black;
  background-color: rgba(220, 220, 220, 0.564);
  margin-left: 2rem;
  margin-top: .7em;
  width: 12rem;
}

.btn {
  &,
  &:link,
  &:visited {
    text-transform: uppercase;
    text-decoration: none;
    padding: 1.5rem 2.5rem;
    border-radius: 10rem;
    display: inline-block;
    transition: all .2s;
    border: none;
    cursor: pointer;
    }

  &:hover {
    transform: translateY(-2px);
    background-color: rgba($color-white, 0.023);
    box-shadow: 0 1rem 2em rgba(0
    ,0,0,.2);
    }

  &:focus,
  &:active {
    transform: translateY(-1px);
    box-shadow: 0 .5rem 1rem rgba(0
    ,0,0,.2);
  }
}

.loginForm {
  width: 85vw;
  height: 55vh;
  font-family: $font-Bebas;

  &__label {
     font-family: $font-Bebas;
     color: #333;
     -webkit-font-smoothing: antialiased;
     -moz-osx-font-smoothing: grayscale;
     cursor: pointer;
  }
}

.heading {
  font-size: 3rem;
  font-weight: 400;
  margin-left: 2rem;
  margin-top: .7em;
}

.section-form {
  padding: 15rem 0;
  background-Image: linear-gradient( to right bottom, $color-white, $color-black);
  background-size: cover;
  background-repeat: no-repeat;
}


.user-info {
  background-image: linear-gradient(105deg,
                                    rgb( 255, 255, 255) 0%,
                                    rgba(255, 255, 255, .9) 70%,
                                     transparent 100%);
  border-radius: 3px;
  box-shadow: 0 1.5rem 4rem rgba($color-black, .15);
  width: 95rem;
  height: 55rem;
  margin: auto;
  &__form {
    width: 50%;
    padding: 3rem;

  }
}


.loginForm {
  &__group:not(:last-child) {
    margin-bottom: 2rem;

  }
  &__input {
    font-family: inherit;
    font-size: 1.5rem;
    color: $dark-grey-color;
    padding: 1.5rem 2rem;
    border-radius: 2px;
    background-color: rgba( 255, 255, 255, .5);
    border: none;
    border-bottom: 3px solid transparent;
    width: 40%;
    display:block;
    transition: all .3s;

   &::-webkit-input-placeholder {
    color: $color-dark-grey-2;
    }

   &:focus {
    outline:none;
    box-shadow: 0 1rem 2rem rgba(0, 0, 0, .1);
    border-bottom: 3px solid $primary-color;
    }

  &:focus:invalid {
    border-bottom: 3px solid $color-secondary-dark;
    }

  }

  &__label {
    font-size: 1.2rem;
    font-weight: 400;
    margin-left: 2rem;
    margin-top: .7em;
    display: block;
    transition: all .3s;
  }

  &__input:placeholder-shown + &__label {
    opacity: 0;
    visibility: hidden;
    transform: translateY(-4rem);
  }

  &__group {
    margin-left: 2rem;
    margin-top: .7em
  }
}

.moviesContainer {
  display: flex;
  flex-direction: row;
}

.moviesContainer-movieCard-cardContainer {
  display: flex;
  flex-direction: column;
  flex-direction: column;
  justify-content: center;
  height: 100%;
}

.movieCard-cardContainer-title {
  font-size: 2em;
}

.movieCard-cardContainer-poster {
  align-items: center;
  height: 300px;
  width: 250px;
}