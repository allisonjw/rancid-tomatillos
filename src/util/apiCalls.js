export const getMovieData = async (url) => {
    const response = await fetch(url)
    const data = await response.json()
    return data;
};


export const fetchUser = async (email, password) => {
    const userLogin = { email: `${email}`, password: `${password}`};
    const options = {
        method: 'POST',
        body: JSON.stringify(userLogin),
        headers: {
            'Content-Type': 'application/json'
        }
    };

    const response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v1/login', options)
    if(!response.ok){
        throw new Error('oops! please check your username and password are correct. ')
    }
    return response.json();
};

export const postRating = async (userRating, userID) => {
  const options = {
      method: 'POST',
      body: JSON.stringify(userRating),
      headers: {
          'Content-Type': 'application/json'
      }
  };

  const post = await fetch(`https://rancid-tomatillos.herokuapp.com/api/v1/users/${userID}/ratings`, options)
  const rating = await post.json()
  return rating;
};

export const fetchRatings = userID => {
    return fetch(`https://rancid-tomatillos.herokuapp.com/api/v1/users/${userID}/ratings`).then(response => {
      if (!response.ok) {
        throw Error('There was a problem fetching all ratings');
      }
      return response.json();
    });
};


export const removeRating = async (userID, ratingID) => {
    const url = `https://the-screening-room-db.herokuapp.com/api/v1/users/${userID}/moviefavorites/${ratingID}`;
        const options = {
            method: "DELETE",
            headers: {
                "Content-Type": "application/json"
            }
        };
        const response = await fetch(url, options);
        if (!response.ok) {
            throw Error('There was changing your rate')
        }
        return response.json()

};
