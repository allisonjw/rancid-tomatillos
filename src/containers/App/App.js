import React, { Component } from 'react';
// import LoginForm from '../LoginForm/LoginForm';
import { getMovieData } from '../../util/apiCalls/apiCalls'
import UserProfile from '../UserProfile/UserProfile';
import MoviesContainer from '../MoviesContainer/MoviesContainer';
// import ShowPage from '../../components/ShowPage/ShowPage';
// import MovieRatings from '../MovieRatings/MovieRatings';
import './App.scss';

export default class App extends Component {
  constructor() {
    super()
    this.state = {

    }
  }

  componentDidMount() {
    getMovieData('https://rancid-tomatillos.herokuapp.com/api/v1/movies')
    .then(data => console.log(data))
  }

  render() {
    return (
      <main className="app-main">
        {/* <LoginForm /> */}
        <UserProfile />
        <MoviesContainer />
        {/* <ShowPage /> */}
        {/* <MovieRatings /> */}
      </main>
    )
  }
}