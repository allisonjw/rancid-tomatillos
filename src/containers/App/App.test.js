import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { App } from './App';
import { getMovieData } from '../../util/apiCalls.js';
import { fetchUser } from '../../util/apiCalls.js';

jest.mock('../../util/apiCalls.js');

describe('App', () => {


  beforeEach(() => {

    getMovieData.mockImplementation(() => {
      return Promise.resolve({ movies:
        [
          {
            id: 1,
            title: "Squirrelzilla: The Reckoning"
          },
          {
            id: 2,
            title: "Child's Play IX: Elf on the Shelf"
          }
        ]
      })
    })

  })

  it('should render with correct data', () => {
    const wrapper = shallow(
      <App addMovies={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot()
  })

  it('should call getMovieData when mounted', () => {
    shallow( <App addMovies={jest.fn()}/>)
    expect(getMovieData).toHaveBeenCalled()
  })

  it('should call addMovies when mounted', async () => {
    const wrapper = shallow(
      <App addMovies={jest.fn()}/>);

      let mockMovies= { movies:
        [
          {
            id: 1,
            title: "Squirrelzilla: The Reckoning"
          },
          {
            id: 2,
            title: "Child's Play IX: Elf on the Shelf"
          }
        ]
      }

    await wrapper.instance().componentDidMount()

    expect(wrapper.instance().props.addMovies).toHaveBeenCalledWith(mockMovies)
  })
})
