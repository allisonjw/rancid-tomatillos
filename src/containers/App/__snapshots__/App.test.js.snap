// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render with correct data 1`] = `
<main
  className="app-main"
>
  <Route
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/movies/:id"
    render={[Function]}
  />
</main>
`;
